pkg/account/deployer/client.go

```diff
@@ -218,7 +218,7 @@ func (d *accountManagementClient) updatePermissions(ctx context.Context, groupId
 		return fmt.Errorf("group id must not be empty")
 	}
 
-	if len(permissions) == 0 {
```

Comment:
basically, this check prevented the flow from continuing, resulting in no updates happening when there are no references at all. 
same for other occurences.